### DISCUSSION TITLE
Different behavior with and without config file?

### DISCUSSION THREAD

**Discussion by ntersol-tom** [BurntSushi/ripgrep#2339] [Q&A] [answered]
What am I doing wrong? (on macOS with zsh)

$ alias rg

$ cat $RIPGREP_CONFIG_PATH
--glob='!.git'
--hidden
--smart-case

$ rg terminess
No files were searched, which means ripgrep probably applied a filter you didn't expect.
Running with --debug will show why files are being skipped.

$ rg terminess .

$ RIPGREP_CONFIG_PATH= rg --glob='!.git' --hidden --smart-case terminess
tvaughan/.emacs.d/init.local.el
87:(setq radian-font "TerminessTTF Nerd Font Mono")

tvaughan/.Brewfile
48:cask 'font-terminessttf-nerd-font'

$ rg --version
ripgrep 13.0.0
-SIMD -AVX (compiled)

$ zsh --version
zsh 5.9 (arm-apple-darwin21.3.0)

**Comment by BurntSushi** [comment, marked as answer]
It's because your config file and your shell command are not equivalent. The config file is just a file. But your command is executed in the context of the shell that understands quoting and all sorts of things. So in your config file, you shouldn't be using quotes. Just `--glob=!.git` will do. Or equivalently:

```
--glob
!git
```

**Comment by ntersol-tom** [comment]
That did the trick. Thanks @BurntSushi
